src/impl.cpp

```diff
@@ -0,0 +1,11 @@
+#include "impl.h"
+
+char* IntToBinaryRepresentation(char* bufer, int source) {
+    constexpr int sz = sizeof(int) *8;
+    for (int i=0; i< sz; i++) {
+        bufer[sz-i -1]= '0' + source ^ ((source >> 1) << 1);
```

Comment:
What the profit of moving right and back then? Wouldn't it be easier to understand/implement this method without XOR?
Think how it could be reworked without XOR
Also, I'm not sure that this will work for another cases, please check this solution with additional test cases